### ISSUE TITLE
check for keys doesn't account for @@archive

### ISSUE THREAD

**Issue by cperciva** [Tarsnap/tarsnap#62]
When we're checking to make sure we have all the keys we need, we're assuming that archive creation only requires write keys.  But if there's an @@archive directive, we will need read keys as well, which means we could end up at "Programmer error: key not available".

I'm not sure what the best solution is for this; scanning the command line for @@archive directives?  We need to be careful to not trigger on "-C @@foo" in case someone has a peculiarly named directory though.  We could postpone the check until we actually hit the @@archive directive, but it would be much better if we could inform the user that they're using the wrong key file promptly...

**Comment by gperciva** [comment]
Note to self: assuming you have the file `ok.txt` in archive `a0`, reproduce with:
```
./tarsnap -c --dry-run --include "ok.txt" @@a0
```

**Comment by gperciva** [comment]
Correction for testing:
```
mkdir @@blah
touch @@blah/foo.txt
./tarsnap -c --dry-run @@a0 -C @@blah foo.txt
```

**Comment by gperciva** [comment]
Should be addressed in #68.
